agent/command/attach_artifacts.go

```diff
@@ -80,7 +86,7 @@ func (c *attachArtifacts) Execute(ctx context.Context,
 	for idx := range c.Files {
 		segment, err = readArtifactsFile(getWorkingDirectory(conf, c.Prefix), c.Files[idx])
 		if err != nil {
-			if c.Optional && os.IsNotExist(errors.Cause(err)) {
+			if (c.Optional || c.ExactFileNames) && os.IsNotExist(errors.Cause(err)) {
```

Comment:
Why do we want to ignore the errors for exactFileNames? I don't know if exactFileNames should automatically be considered optional.

Reply:
I put this to make it such that the command won't fail if the files specified in the YAML don't exist, since `ExactFileNames` is explicit about what files to pick and the current way the command works is more forgiving in this regard since it automatically detects files based on the gitignore style matching. 

That being said I'm fine to remove this and have the expectation that the user is going to be sure the files with these exact names are going to exist or else it will fail.

Reply:
Since this request came from a specific user, I'd suggest consulting them to see which they prefer. 
If they prefer that if behave like optional does, I'd suggest adding that to the docs.